main.py

```diff
@@ -41,99 +40,175 @@ def check(a, b):
     """Updates the given board and returns it; puts NextState values in CurrentState"""
-    for a in range(config.Width + 2 * config.Cushion):
-        for b in range(config.Height + 2 * config.Cushion):
-            board[a][b].CurrentState = board[a][b].NextState
+    for a in range(board.Width + 2 * board.Cushion):
+        for b in range(board.Height + 2 * board.Cushion):
+            board.Cell[a][b].CurrentState = board.Cell[a][b].NextState
     return board
 
 
+def draw_gps_slider(y, gps_limit):
+    """Draws the slider with the y coordinate of the button click
+       (How many GPS this corresponds to is not dealt with here.)"""
+    if y < Widgets.StartOfSlider:
+        y = Widgets.StartOfSlider
+    elif y > Widgets.EndOfSlider:
+        y = Widgets.EndOfSlider
+    pygame.draw.rect(Screen, GameState.Colour["Background"], ((Widgets.ButtonStart, Widgets.StartOfSlider - Widgets.NotchLength),
+                                                 (Widgets.ButtonStart + Board.Edge + Widgets.HighlightSize +
+                                                  Widgets.ButtonSize, Widgets.EndOfSlider)))
+    pygame.draw.line(Screen, GameState.Colour["Text"], (Widgets.SliderY, Widgets.StartOfSlider), (Widgets.SliderY,
+                                                                                       Widgets.EndOfSlider))
+    for n in range(Widgets.NoOfNotches):
+        pygame.draw.line(Screen, GameState.Colour["Text"], (Widgets.SliderY - Widgets.NotchLength / 2,
+                                                   Widgets.StartOfSlider + n * Widgets.SpaceBetweenNotches),
+                         (Widgets.SliderY + Widgets.NotchLength / 2,
+                          Widgets.StartOfSlider + n * Widgets.SpaceBetweenNotches))
+    config.write(Screen, Widgets.SliderY - (12 + Widgets.NotchLength), (Widgets.StartOfSlider + Widgets.EndOfSlider) * 0.5,
+                 "Speed", GameState.Colour["Text"], 20,
+                 rotate=90, alignment=("left", "centre"))
+    if gps_limit:
+        colour = "Highlighter"
+    else:
+        colour = "Unselected"
+    pygame.draw.polygon(Screen, GameState.Colour[colour], ((Widgets.SliderY + Widgets.NotchLength / 2, y),
+                                         (Widgets.SliderY + Widgets.NotchLength, y - Widgets.NotchLength / 2),
+                                         (Widgets.SliderY + 2 * Widgets.NotchLength, y - Widgets.NotchLength / 2),
+                                         (Widgets.SliderY + 2 * Widgets.NotchLength, y + Widgets.NotchLength / 2),
+                                         (Widgets.SliderY + Widgets.NotchLength, y + Widgets.NotchLength / 2)))
+    pygame.display.update()
+
+
 pygame.init()
-Screen = pygame.display.set_mode((config.Size * config.Width, config.Size * config.Height))
-Screen.fill(config.Background)
-pygame.display.set_caption("Game of Life")
-Board = clean_board()
-Paused = True
+pygame.event.set_allowed(None)
+pygame.event.set_allowed(pygame.MOUSEMOTION)
```

Comment:
Cut the copypasta in a separate file/function

Reply:
made it into a list